onebusaway-transit-data-federation/src/main/java/org/onebusaway/transit_data_federation/impl/beans/RouteScheduleBeanServiceImpl.java

```diff
@@ -186,41 +188,35 @@ private void addStopTripDirectionsViaBlockTrip(RouteScheduleBean rsb, AgencyAndI
           } else {
             headsign = getDestinationForTrip(blockTrip.getTrip());
           }
-          DirectionalHeadsign dh = new DirectionalHeadsign(directionId, headsign);
-          if (!headsignToBeanMap.containsKey(dh)) {
+          StopCollection stops = new StopCollection();
+          stops.addFromTrip(blockTrip.getTrip());
+          DirectionHeadsignStopCollection directionHeadsignStopCollection =
+                  new DirectionHeadsignStopCollection(directionId,headsign,stops);
+          Double dirHeadStopHashVal = directionHeadsignStopCollection.getHash();
+          //Should we be protecting these hashmaps from false matches?
```

Comment:
Comments like these are scary.  Either convince yourself it's correct or ask someone to review.